packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -74,20 +75,151 @@ import {
+import { FluidDataStoreRegistry } from "./dataStoreRegistry.js";
+
+/**
+ * Accepted header keys for requests coming to the runtime.
+ * @internal
+ */
+export enum RuntimeHeaders {
+	/** True to wait for a data store to be created and loaded before returning it. */
+	wait = "wait",
+	/** True if the request is coming from an IFluidHandle. */
+	viaHandle = "viaHandle",
+}
+
+/** True if a tombstoned object should be returned without erroring
+ * @alpha
+ */
+export const AllowTombstoneRequestHeaderKey = "allowTombstone"; // Belongs in the enum above, but avoiding the breaking change
+/**
+ * [IRRELEVANT IF throwOnInactiveLoad OPTION NOT SET] True if an inactive object should be returned without erroring
+ * @internal
+ */
+export const AllowInactiveRequestHeaderKey = "allowInactive"; // Belongs in the enum above, but avoiding the breaking change
 
 type PendingAliasResolve = (success: boolean) => void;
 
+interface FluidDataStoreMessage {
+	content: any;
+	type: string;
+}
+
+export function wrapContext(context: IFluidParentContext): IFluidParentContext {
+	return {
+		get IFluidDataStoreRegistry() {
+			return context.IFluidDataStoreRegistry;
+		},
+		IFluidHandleContext: context.IFluidHandleContext,
+		options: context.options,
+		get clientId() {
+			return context.clientId;
+		},
+		get connected() {
+			return context.connected;
+		},
+		deltaManager: context.deltaManager,
+		storage: context.storage,
+		logger: context.logger,
+		get clientDetails() {
+			return context.clientDetails;
+		},
+		idCompressor: context.idCompressor,
+		loadingGroupId: context.loadingGroupId,
+		get attachState() {
+			return context.attachState;
+		},
+		containerRuntime: context.containerRuntime,
+		scope: context.scope,
+		gcThrowOnTombstoneUsage: context.gcThrowOnTombstoneUsage,
+		gcTombstoneEnforcementAllowed: context.gcTombstoneEnforcementAllowed,
+		getAbsoluteUrl: async (...args) => {
+			return context.getAbsoluteUrl(...args);
+		},
+		getQuorum: (...args) => {
+			return context.getQuorum(...args);
+		},
+		getAudience: (...args) => {
+			return context.getAudience(...args);
+		},
+		ensureNoDataModelChanges: (...args) => {
+			return context.ensureNoDataModelChanges(...args);
+		},
+		submitMessage: (...args) => {
+			return context.submitMessage(...args);
+		},
+		submitSignal: (...args) => {
+			return context.submitSignal(...args);
+		},
+		makeLocallyVisible: (...args) => {
+			return context.makeLocallyVisible(...args);
+		},
+		uploadBlob: async (...args) => {
+			return context.uploadBlob(...args);
+		},
+		addedGCOutboundReference: (...args) => {
+			return context.addedGCOutboundReference?.(...args);
+		},
+		getCreateChildSummarizerNodeFn: (...args) => {
+			return context.getCreateChildSummarizerNodeFn?.(...args);
+		},
+		deleteChildSummarizerNode: (...args) => {
+			return context.deleteChildSummarizerNode?.(...args);
+		},
+	};
+}
+
+/** @internal */
+export function createParentContext(
```

Comment:
Consider renaming this function to better represent that it is for channel in channel collection.
Unrelated - but should we rename DataStores class to ChannelCollection class? (Navin's idea)

Reply:
I would rename it but maybe in the next PR, I don't want to make this larger than it already is..